Template enhancements

DerAndereAndi opened the pull request:
- Only add SMA HEMS Setup, if required values exist (e.g. machine-id). Also asks the user if this should be added
- New templates: Alfen, PC Electric Garo, go-e V3 (moved into separate template), Tasmota meter, Huawei inverter with Sdongle
- Updated templates: SolarEdge Inverter, Sunspec
- New capabilities can be assigned: 1p3p, RFID
- Added group information instead of `generic` flag. Used e.g. for marking power plugs or generic devices
- Requirement description text can now be markdown text (required for documentation page)
- Replaced `description` with product list, separating brand from description (required for documentation page)
- Generate newly structured documentation yaml files
- ...

Fix https://github.com/evcc-io/evcc/issues/2385

DerAndereAndi committed:
Only add SMA HEMS if all required values exist

Also SMA HEMS is now a capability and not a requirement

DerAndereAndi committed:
Moev go-e V3 into a separate template

DerAndereAndi committed:
Add Alfen template

DerAndereAndi committed:
Add 1p3p phase switching capability information

DerAndereAndi committed:
Replace template description with products list

- Allow to define a list of products this template supports
- Allow to define a brand per product

DerAndereAndi committed:
Replace generic with devicegroup reference

This allows to create groups of all kinds of devices, defined in `devicegroups.yaml`. Today we have `generic` (e.g. EEBUS, SunSpec, ...) and `sockets` for power sockets and `switchsockets` for switchable sockets

DerAndereAndi committed:
Add missing generated docs

DerAndereAndi committed:
Add tasmota meter

DerAndereAndi committed:
Rename devicegroup to group and fix sorting

DerAndereAndi committed:
Add RFID capability

Not directly used yet

DerAndereAndi committed:
Update product structure

DerAndereAndi committed:
Support for markdown in requirement description

DerAndereAndi committed:
Multiple updates

- Generate new documentation files
- Update `requirements` and `capabilities` structures
- Added a few more help texts

andig commented:
@DerAndereAndi was ist an der v3 aus Templatesicht anders?

DerAndereAndi commented:
- 1p3p Support (kann in der Doku angezeigt werden)
- Requirements, also HTTP API und Firmware

Render ist  gleich, aber die Meta Informationen sind anders. daher haben wir das getrennt. Für die Strukturierung der Templates sollte nicht nur die Implementierung maßgebend sein, sondern auch die Verwendung und die dazu notwendigen Informationen.

DerAndereAndi committed:
Add more validation checks

DerAndereAndi committed:
Add serial param to Keba

DerAndereAndi committed:
Move global param defaults into yaml

naltatis commented on templates/docs/meter/kostal-piko_0.yaml:
```diff
@@ -0,0 +1,27 @@
+    type: template
+    template: kostal-piko
+    usage: grid
+    host: 192.0.2.2  # IP-Adresse oder Hostname 
```
Würde vorschlagen, dass die `advanced` Version nur im yaml steht, wenn es auch wirklich erweiterte Einstellungen gibt. Sonst würde ein `weitere Optionen anzeigen` Toggle ja zu keinem Unterschied führen.

DerAndereAndi replied:
ich schau mal wie ich das hinbekomme

DerAndereAndi replied:
Done

naltatis commented on templates/definition/vehicle/vwid.yaml:
```diff
@@ -1,6 +1,8 @@
+products:
+- brand: Volkswagen
+  description:
+    generic: We Connect ID
```
👍 Hier passt die Brand/Description Struktur auch total gut. Gefällt mir.

naltatis commented on templates/docs/vehicle/tesla_0.yaml:
```diff
@@ -0,0 +1,26 @@
+    template: tesla
+    title: # Wird in der Benutzeroberfläche angezeigt  # Optional 
+    accessToken: # Siehe https://docs.evcc.io/docs/devices/vehicles#tesla 
+    refreshToken: # Siehe https://docs.evcc.io/docs/devices/vehicles#tesla 
```
"Siehe https://docs ..." Müssten wir alle durch den aktuellen manuell gepflegten Inhalt im Wiki ersetzen.

DerAndereAndi replied:
Done

naltatis commented on templates/definition/meter/sungrow.yaml:
```diff
@@ -1,6 +1,7 @@
-  generic: Sungrow (Beta)
+products:
+- description:
+    generic: Sungrow (Beta)
```
Ist "Sungrow" nicht die Brand? Aktuell gibts vermutlich die Logik, dass es immer eine `description` geben muss. Evtl. könnten wir das zu "mind. brand o. description" verändern?

DerAndereAndi replied:
Gibt es Hersteller ohne Produkte? :D Bei der Konfig wissen wir sowieso noch nicht ob die funktioniert. Aber wohl dann eher brand `Sungrow` und als description dann `Inverter`

DerAndereAndi replied:
Done

DerAndereAndi committed:
Only show advanced section if it has such params

DerAndereAndi committed:
Update sungrow template

DerAndereAndi committed:
Simplify

DerAndereAndi committed:
Update requirements text

DerAndereAndi committed:
Update tesla template

DerAndereAndi committed:
Quite a big update

- Moved `grouplist.yaml`, `parambaselist.yaml` and `paramdefaults.yaml` into `defaults.yaml`
- Added central params definition in `defaults.yaml` with param specific localization, hence that is removed from the language files of the cli
- Renamed `base` param into `preset`
- Remove most hardcoded modbus handling (`Template_modbus.go` is missing)
- Add `modbus` definitions into `defaults.yaml`

DerAndereAndi committed:
Fix evnotify template

DerAndereAndi committed:
Merge branch 'master' into feature/template-enhancements

DerAndereAndi committed:
Fix param dependency checks

DerAndereAndi committed:
Add evsewifi error for test

DerAndereAndi committed:
Add PC Electric Garo template

DerAndereAndi committed:
Add missing generated doc

DerAndereAndi committed:
Various fixes and updates

DerAndereAndi committed:
Update Garo template

DerAndereAndi committed:
Fix linter

DerAndereAndi committed:
Updated generated doc

DerAndereAndi committed:
Merge branch 'master' into feature/template-enhancements

DerAndereAndi committed:
Update smart tempalte

DerAndereAndi committed:
Add Huawei Sdongle templates

DerAndereAndi committed:
Improve sponsorship handling at end of flow

DerAndereAndi committed:
Update solardge inverter and sunspec template

DerAndereAndi committed:
Remove SMA HEMS battery check

naltatis commented on templates/docs/vehicle/nissan_0.yaml:
```diff
@@ -0,0 +1,27 @@
+product:
+  brand: Nissan
```
Es gibt aktuell zwei Nissan Fahrzeuge. Dieser hier und Carwings. Wenns mehr als einen gibt sollten wir immer auch das `description` Feld füllen damit wir in der Doku Unterüberschriften haben.

![Bildschirmfoto 2022-01-16 um 23 34 08](https://user-images.githubusercontent.com/152287/149680866-faf747e9-96b5-4b6c-b7b2-2fdcfac59c63.png)

DerAndereAndi replied:
Done

DerAndereAndi committed:
Merge branch 'master' into feature/template-enhancements

DerAndereAndi committed:
Remove wrong link in alphatec charger

DerAndereAndi committed:
Add Alphatec template

DerAndereAndi committed:
Update nissan template

DerAndereAndi committed:
Update Huawei dongle template

DerAndereAndi committed:
Handle Kostal Piko strange grid power handling

It reports power 0 and no error even if there is no grid meter attached.

andig commented on charger/alphatec.go:
```diff
@@ -27,8 +27,6 @@ import (
 )
 
-// https://github.com/evcc-io/evcc/discussions/1965
-
```
Warum diese Änderung?

DerAndereAndi replied:
Der Link hat nichts mit dem Template zu tun.

DerAndereAndi replied:
Doku der Alfen Wallbox. Ist das identisch zur Alphatec? Wieso sind das dann 2 Implementierungen? Verstehe ich gerade nicht

andig commented on templates/definition/charger/alfen.yaml:
```diff
@@ -0,0 +1,14 @@
+  evcc: ["sponsorship"]
+params:
+- name: modbus
+  choice: ["rs485"]
```
TCP!

DerAndereAndi committed:
Fix modbus type

DerAndereAndi committed:
Add Wahlter Werke Basic Evo Pro

DerAndereAndi committed:
Merge branch 'master' into feature/template-enhancements

DerAndereAndi committed:
Update mercedes template

DerAndereAndi committed:
Support for #2386
